tools/shared.py

```diff
@@ -772,6 +772,54 @@ def clean_temp():
     prepare_to_clean_temp(EMSCRIPTEN_TEMP_DIR) # this global var might change later
   return EMSCRIPTEN_TEMP_DIR
 
+class Enum(set):
+  def __getattr__(self, key):
+    if key in self:
+      return key
+
+WarningType = Enum(['ABSOLUTE_PATHS', 'SEPARATE_ASM', 'ALMOST_ASM'])
```

Comment:
I like that idea; the warning manager has default warning text, which can be overridden (or appended to?) by the caller.